assignments/closure.js

```diff
@@ -3,8 +3,19 @@
 
 
 // ==== Challenge 2: Create a counter function ====
+let rock = "Rocks are Cool";
+  function pinkDiamond() {
+  return rock;
+}
+
+console.log(pinkDiamond);
 const counter = () => {
   // Return a function that when invoked increments and returns a counter variable.
+  let count = 0;
```

Comment:
A few things here. Watch your return statement. You have assigned the anonymous function that returns ++count to the variable addOne. You would have to run the return function of counter `counter()()` to get the intended return. Also, you are declaring your variable inside of your counter function and setting it to 0. So every time you run the counter function the first thing that happens is counter is set to 0. So no matter how many times you run `counter()()` in a row, the outcome will always be 1. What I would do is get rid of the outer function here. Keep count on the global scope and make addOne your counter function. By keeping count outside of your function closure, you could increment every time you ran that function. three `addOne()` and the third one will return 3 and count will have the value of 3.